tree/tree/test/BulkApiMultiple.cxx

```diff
@@ -25,27 +25,223 @@ class BulkApiMultipleTest : public ::testing::Test {
 
       // Otherwise, we keep with the current ROOT defaults.
       auto tree = new TTree("T", "A ROOT tree of floats.");
+      // bool, short, int, float, int64_t to work, while "char
       float f = 2;
       double g = 3;
+      bool b = true;
+      short s  = 4;
+      int i = 5;
+      Long64_t ll = (1<<20) + 1;
+      char c = 7;
+
       TBranch *branch2 = tree->Branch("myFloat", &f, 320000, 1);
       TBranch *branch3 = tree->Branch("myDouble", &g, 320000, 1);
+      TBranch *branch4 = tree->Branch("myBool", &b, 320000, 1);
+      TBranch *branch5 = tree->Branch("myShort", &s, 320000, 1);
+      TBranch *branch6 = tree->Branch("myInt", &i, 320000, 1);
+      TBranch *branch7 = tree->Branch("myLongLong", &ll, 320000, 1);
+      TBranch *branch8 = tree->Branch("myChar", &c, 320000, 1);
+
       branch2->SetAutoDelete(kFALSE);
       branch3->SetAutoDelete(kFALSE);
+      branch4->SetAutoDelete(kFALSE);
+      branch5->SetAutoDelete(kFALSE);
+      branch6->SetAutoDelete(kFALSE);
+      branch7->SetAutoDelete(kFALSE);
+      branch8->SetAutoDelete(kFALSE);
+
       for (Long64_t ev = 0; ev < fEventCount; ev++) {
          tree->Fill();
          f ++;
          g ++;
+         b = !b;
+         s ++;
+         i ++;
+         ll ++;
+         c ++;
       }
       hfile = tree->GetCurrentFile();
       hfile->Write();
-      tree->Print();
+      //tree->Print();
       printf("Successful write of all events.\n");
       hfile->Close();
 
       delete hfile;
    }
 };
 
+template <typename T>
+void increment(T &value)
+{
+   value++;
+}
+
+void increment(bool &value)
+{
+   value = !value;
+}
+
+template <typename T>
+void SimpleBulkReadFunc(const char *filename, const char *treename, const char *branchname, T initialvalue)
+{
+   auto hfile = TFile::Open(filename);
+   printf("Starting read of file %s.\n", filename);
+   TStopwatch sw;
+
+   printf("Using outlined bulk read APIs.\n");
+   TBufferFile branchbuf(TBuffer::kWrite, 32 * 1024);
+   auto tree = hfile->Get<TTree>(treename);
+   ASSERT_TRUE(tree);
+
+   TBranch *branch = tree->GetBranch(branchname);
+   ASSERT_TRUE(branch);
+   //branch->GetListOfBaskets()->ls();
+
+   T value = initialvalue;
+   Long64_t evt_idx = 0;
+   Long64_t events = tree->GetEntries();
+   while (events) {
+      auto count = branch->GetBulkRead().GetBulkEntries(evt_idx, branchbuf);
+      ASSERT_GE(count, 0);
+      events = events > count ? (events - count) : 0;
+
+      auto entry = reinterpret_cast<T *>(branchbuf.GetCurrent());
+      for (Int_t idx = 0; idx < count; idx++) {
+         if (R__unlikely((evt_idx < 16000000) && (entry[idx] != value))) {
+            std::cerr << "In tree " << treename << " Incorrect value for " << branchname
+                      << " branch: " << entry[idx] << " instead of " << value << " at entry #" << evt_idx + idx << '\n';
+            tree->Scan(branchname, "", "", 10, evt_idx + idx);
+            ASSERT_TRUE(false);
```

Comment:
```suggestion
         EXPECT_TRUE(evt_idx < 16000000 || entry[idx] == value)
            << "In tree " << treename << " Incorrect value for " << branchname
            << " branch: " << entry[idx] << " instead of " << value << " at entry #" << evt_idx + idx
            << tree->Scan(branchname, "", "", 10, evt_idx + idx), '';
```